include/measurement_kit/nettest.hpp

```diff
@@ -0,0 +1,2589 @@
+class StatusMeasurementSubmissionEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "status.measurement_submission";
+    
+    /// The "idx" attribute of this event.
+    int64_t idx = 0;
+};
+
+/// C++ representation of the "status.measurement_done" event.
+class StatusMeasurementDoneEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "status.measurement_done";
+    
+    /// The "idx" attribute of this event.
+    int64_t idx = 0;
+};
+
+/// C++ representation of the "status.report_close" event.
+class StatusReportCloseEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "status.report_close";
+    
+    /// The "report_id" attribute of this event.
+    std::string report_id = "";
+};
+
+/// C++ representation of the "status.report_create" event.
+class StatusReportCreateEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "status.report_create";
+    
+    /// The "report_id" attribute of this event.
+    std::string report_id = "";
+};
+
+/// C++ representation of the "status.resolver_lookup" event.
+class StatusResolverLookupEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "status.resolver_lookup";
+    
+    /// The "ip_address" attribute of this event.
+    std::string ip_address = "";
+};
+
+/// C++ representation of the "status.started" event.
+class StatusStartedEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "status.started";
+    /* No attributes */
+};
+
+/// C++ representation of the "status.update.performance" event.
+class StatusUpdatePerformanceEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "status.update.performance";
+    
+    /// The "direction" attribute of this event.
+    std::string direction = "";
+    
+    /// The "elapsed" attribute of this event.
+    double elapsed = 0.0;
+    
+    /// The "num_streams" attribute of this event.
+    int64_t num_streams = 0;
+    
+    /// The "speed_kbps" attribute of this event.
+    double speed_kbps = 0.0;
+};
+
+/// C++ representation of the "status.update.websites" event.
+class StatusUpdateWebsitesEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "status.update.websites";
+    
+    /// The "url" attribute of this event.
+    std::string url = "";
+    
+    /// The "status" attribute of this event.
+    std::string status = "";
+};
+
+/// C++ representation of the "task_terminated" event.
+class TaskTerminatedEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "task_terminated";
+    /* No attributes */
+};
+
+} // namespace events
+
+#if !defined SWIG && !defined DOXYGEN
```

Comment:
Internal undocumented namespace containing stuff that must be here for the remainder of the code to work but that generally one should not look into. Maybe put this instead inside `ffi.h`?